src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,97 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.InvalidUserDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(null);
+        });
+        assertEquals("User cannot be null", exception.getMessage());
+    }
+
+    @Test
+    void register_userWithExistingLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("testuser");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("testuser");
+        user2.setPassword("newpassword");
+        user2.setAge(22);
+
+        Exception exception = assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user2);
+        });
+        assertEquals("User with this login already exists", exception.getMessage());
+    }
```

Comment:
This test case checks if a user with an existing login can be registered again. However, the test case does not verify if the user was added to the storage after successful registration. This can be done by asserting that the size of the storage has increased by 1 after the registration.